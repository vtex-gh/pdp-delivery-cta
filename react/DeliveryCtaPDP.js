import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
