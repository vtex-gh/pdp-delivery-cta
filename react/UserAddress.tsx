import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
