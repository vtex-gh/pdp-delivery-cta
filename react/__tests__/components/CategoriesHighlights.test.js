import React from 'react'
import { render } from '@vtex/test-tools/react'

import CategoriesHighlights from '../../CategoriesHighlights'

describe('<CategoriesHighlights />', () => {
  const renderComponent = customProps => {
    const props = {
      cardShape: 'squared',
      showCategoriesHighlighted: true,
      ...customProps,
    }

    return render(<CategoriesHighlights {...props} />)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot Squared', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot Rectangular', () => {
    const { asFragment } = renderComponent({ cardShape: 'rectangular' })

    expect(asFragment()).toMatchSnapshot()
  })
})
