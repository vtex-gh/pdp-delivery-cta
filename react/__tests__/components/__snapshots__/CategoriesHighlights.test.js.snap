// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoriesHighlights /> should match snapshot Rectangular 1`] = `
<DocumentFragment>
  <div
    class="rectangularCategoriesHighlights relative"
  >
    <div
      class="flex flex-row flex-wrap items-center justify-center"
    >
      <div
        class="flex flex-row flex-wrap items-center justify-center"
      >
        <div
          class="rectangularCard shadow-1 ma1"
        >
          <a
            href="dummy"
          >
            <img
              alt=""
              class="rectangularCardImage"
              src="rectangular-placeholder.svg"
            />
          </a>
        </div>
        <div
          class="rectangularCard shadow-1 ma1"
        >
          <a
            href="dummy"
          >
            <img
              alt=""
              class="rectangularCardImage"
              src="rectangular-placeholder.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CategoriesHighlights /> should match snapshot Squared 1`] = `
<DocumentFragment>
  <div
    class="squaredCategoriesHighlights relative"
  >
    <div
      class="flex flex-row flex-wrap items-center justify-center"
    >
      <div
        class="flex flex-row flex-wrap items-center justify-center"
      >
        <div
          class="squaredCard shadow-1 ma1"
        >
          <a
            href="dummy"
          >
            <img
              alt=""
              class="squaredCardImage"
              src="squared-placeholder.svg"
            />
          </a>
        </div>
        <div
          class="squaredCard shadow-1 ma1"
        >
          <a
            href="dummy"
          >
            <img
              alt=""
              class="squaredCardImage"
              src="squared-placeholder.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
